packages/sdk-react/src/core/queries/useCounterpart.ts

```diff
@@ -128,251 +95,145 @@ export const counterpartQueryKeys = {
-    },
+        await api.counterparts.getCounterpartsIdBankAccounts.invalidateQueries(
+          queryClient
+        );
+      },
 
-    onError: () => {
-      toast.error(t(i18n)`Failed to update Bank Account.`);
-    },
-  });
+      onError: () => {
+        toast.error(t(i18n)`Failed to update Bank Account.`);
+      },
+    }
+  );
 };
 
 export const useDeleteCounterpartBank = (counterpartId: string) => {
   const { i18n } = useLingui();
-  const { monite } = useMoniteContext();
-  const { remove } = useCounterpartBankListCache(counterpartId);
-
-  return useMutation<void, Error, string>({
-    mutationFn: (bankId) =>
-      monite.api.counterparts.deleteBankAccount(counterpartId, bankId),
-
-    onSuccess: (_, bankId) => {
-      remove(bankId);
+  const { api } = useMoniteContext();
+  const queryClient = useQueryClient();
+
+  return api.counterparts.deleteCounterpartsIdBankAccountsId.useMutation(
+    undefined,
+    {
+      onSuccess: async () => {
+        await api.counterparts.getCounterpartsIdBankAccounts.invalidateQueries(
+          {
+            parameters: { path: { counterpart_id: counterpartId } },
+          },
+          queryClient
+        );
 
-      toast.success(t(i18n)`Bank Account was deleted.`);
-    },
+        toast.success(t(i18n)`Bank Account was deleted.`);
+      },
 
-    onError: () => {
-      toast.error(t(i18n)`Failed to delete Bank Account.`);
-    },
-  });
+      onError: () => {
+        toast.error(t(i18n)`Failed to delete Bank Account.`);
+      },
+    }
+  );
 };
 
 export const useCounterpartVatList = (counterpartId?: string) => {
-  const { monite } = useMoniteContext();
-
-  return useQuery<CounterpartVatIDResponse[], Error>({
-    queryKey: counterpartQueryKeys.vatList(counterpartId),
-
-    queryFn: () =>
-      !!counterpartId
-        ? monite.api.counterparts
-            .getVats(counterpartId)
-            .then((response) => response.data)
-        : [],
+  const { api } = useMoniteContext();
 
-    enabled: !!counterpartId,
-  });
+  return api.counterparts.getCounterpartsIdVatIds.useQuery(
+    {
+      path: {
+        counterpart_id: counterpartId ?? '',
+      },
+    },
+    {
+      enabled: !!counterpartId,
+    }
+  );
 };
 
-export const useCreateCounterpartVat = (counterpartId: string) => {
+export const useCreateCounterpartVat = () => {
   const { i18n } = useLingui();
-  const { monite } = useMoniteContext();
-  const { add } = useCounterpartVatListCache(counterpartId);
-
-  return useMutation<CounterpartVatIDResponse, Error, CounterpartVatID>({
-    mutationFn: (vat) => monite.api.counterparts.createVat(counterpartId, vat),
+  const { api } = useMoniteContext();
 
-    onSuccess: (vat) => {
-      add(vat);
-
-      toast.success(t(i18n)`Vat “${vat.value}” was created.`);
+  return api.counterparts.postCounterpartsIdVatIds.useMutation(undefined, {
+    onSuccess: (counterpart) => {
+      toast.success(t(i18n)`Vat “${counterpart.value}” was created.`);
```

Comment:
Here also, if possible to invalidate, I would do that. But, since there was not in the original code, it's not required.